portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -16,17 +16,57 @@
 
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    private List<String> comments = new ArrayList<String>();
+
+    @Override 
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Gson gson = new Gson();
+        String json = gson.toJson(comments);
+        response.getWriter().println((json));    
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get the input from the form.
+        String comment = getComment(request);
+
+        // Add comment to comments Array.
+        comments.add(comment);
+
+        response.setContentType("text/html;");
+        response.getWriter().println(comments);
+
+        // Redirect back to the HTML page.
+        response.sendRedirect("/index.html");  
+        }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getComment(HttpServletRequest request) {
+    //   Get comment from form.
+    String comment;
+    try{
+        String comment = request.getParameter("comment");
+        }catch(NullPointerException e) { 
+            System.out.print("NullPointerException Caught");
+            return -1; 
```

Comment:
Unlikely loosely typed languages like python and JavaScript, Java requires that methods only return a single type. Since the return type is `String`, -1 can't be used. I'd recommend null to signal to a caller that the comment wasn't able to be retrieved from the request.